docs/spec/generics.rst

```diff
@@ -93,8 +93,13 @@ This is equivalent to omitting the generic notation and just saying
 User-defined generic types
 --------------------------
 
-You can include a ``Generic`` base class to define a user-defined class
-as generic.  Example::
+You can define a user-defined class as generic in three ways: by including a
+``Generic`` base class, by using the  new generic class syntax in Python 3.12
+and higher, or by including a ``Protocol`` base class parameterized with type
+variables. The third approach also marks the class as a protocol - see
+:ref:`generic protocols<generic-protocols>` for more information.
+
+Example using ``Generic``::
```

Comment:
I think it would be good to specify the following rules as well. These are enforced by the runtime, but a type checker should detect and report these error conditions.

If `Generic` is included as a base class, all type parameters must be specified as type arguments to `Generic`.

```python
class Bad1(Iterable[T], Generic): ...  # Error: T does not appear in Generic type arg list
```

If `Protocol` is used as a base class, type parameters can optionally be specified as type arguments to `Protocol`, or `Protocol` can be used without type arguments.

```python
class Bad2(Iterable[T], Protocol[T]): ...  # OK
class Good2(Iterable[T], Protocol): ...  # OK
```

If type parameters are specified using either `Generic` or `Protocol` type arguments, all type parameters for the class must appear within the type argument list. A type checker should report an error if a type variable appears elsewhere in the base class list and that type variable is not included as a type argument for `Generic` or `Protocol`.

```python
class Bad3(Iterable[T], Generic[S]): ...  # Error: T does not appear in Generic type arg list
class Bad4(Iterable[T], Protocol[S]): ...  # Error: T does not appear in Protocol type arg list
```

Reply:
Thanks, these are all great suggestions! I've included them, rephrased slightly to try to fit the flow of the text better. I also noted that all arguments to `Generic` or `Protocol` need to be type variables, since I didn't see that mentioned anywhere.